src/renderer/src/main.tsx

```diff
@@ -7,17 +7,49 @@ import './assets/index.css'
 import { MessageChannelProvider } from './hooks/useMessageChannels'
 import filesTheme from './theme'
 import App from './App'
+import { RecoilSync } from 'recoil-sync'
+
+function RecoilSyncWithDataChannel({ children }: { children: React.ReactNode }) {
+  const [dataChannelPort, setDataChannelPort] = React.useState<MessagePort>()
+
+  React.useEffect(() => {
+    window.onmessage = (event: MessageEvent) => {
+      if (event.source === window && event.data === 'message-channel-ports') {
+        const dataChannelPort = event.ports[2]
+
+        setDataChannelPort(dataChannelPort)
+      }
+    }
+  }, [])
+
+  return (
+    <RecoilSync
+      storeKey="dataChannel"
+      listen={({ updateItem }) => {
+        if (!dataChannelPort) return
+
+        dataChannelPort.onmessage = ({ data: { key, value } }: MessageEvent) => {
+          updateItem(key, value)
+        }
+      }}
+    >
+      {children}
+    </RecoilSync>
+  )
+}
 
 ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
   <React.StrictMode>
     <RecoilRoot>
-      <CssVarsProvider disableTransitionOnChange theme={filesTheme}>
-        <StyledEngineProvider injectFirst>
-          <MessageChannelProvider>
-            <App />
-          </MessageChannelProvider>
-        </StyledEngineProvider>
-      </CssVarsProvider>
+      <RecoilSyncWithDataChannel>
+        <CssVarsProvider disableTransitionOnChange theme={filesTheme}>
+          <StyledEngineProvider injectFirst>
+            <MessageChannelProvider>
+              <App />
+            </MessageChannelProvider>
+          </StyledEngineProvider>
+        </CssVarsProvider>
+      </RecoilSyncWithDataChannel>
     </RecoilRoot>
   </React.StrictMode>
```

Comment:
Here we use Recoil Sync to set up a store with the `dataChannelPort` from the main process.  Currently this is one way communication for the backend main process to message the renderer process and react app.